tutorial/5_sxp_workflow.ipynb

```diff
@@ -279,6 +394,14 @@
     "sp.load_bias_series(binned_data=biasSeries)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### find calibration parameters\n",
+    "We now will fit the tof-energy relation. This is done by finding the maxima of a peak in the tof spectrum, and then fitting the square root relation to obtain the calibration parameters. "
```

Comment:
It plots but also gives. this error when find_bias_peaks is run
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[67], line 3
      1 ranges=(6380, 6700)
      2 ref_id=6
----> 3 sp.find_bias_peaks(ranges=ranges, ref_id=ref_id, apply=True)

File /asap3/fs-flash-o/gpfs/hextof/2024/data/11021313/processed/denoising/sed/sed/core/processor.py:1281, in SedProcessor.find_bias_peaks(self, ranges, ref_id, infer_others, mode, radius, peak_window, apply)
   1278 else:
   1279     # New adjustment tool
   1280     assert isinstance(ranges, tuple)
-> 1281     self.ec.adjust_ranges(
   1282         ranges=ranges,
   1283         ref_id=ref_id,
   1284         traces=self.ec.traces_normed,
   1285         infer_others=infer_others,
   1286         radius=radius,
   1287         peak_window=peak_window,
   1288         apply=apply,
   1289     )

File /asap3/fs-flash-o/gpfs/hextof/2024/data/11021313/processed/denoising/sed/sed/calibrator/energy.py:403, in EnergyCalibrator.adjust_ranges(self, ranges, ref_id, traces, peak_window, apply, **kwds)
    389 refid_slider = ipw.IntSlider(
    390     value=ref_id,
    391     min=0,
    392     max=10,
    393     step=1,
    394 )
    396 ranges_slider = ipw.IntRangeSlider(
    397     value=list(ranges),
    398     min=min(self.tof),
    399     max=max(self.tof),
    400     step=1,
    401 )
--> 403 update(ranges=ranges, refid=ref_id)
    405 ipw.interact(
    406     update,
    407     refid=refid_slider,
    408     ranges=ranges_slider,
    409 )
    411 def apply_func(apply: bool):  # noqa: ARG001

File /asap3/fs-flash-o/gpfs/hextof/2024/data/11021313/processed/denoising/sed/sed/calibrator/energy.py:384, in EnergyCalibrator.adjust_ranges.<locals>.update(refid, ranges)
    381     plot_segs[itr].set_ydata(traceseg)
    382     plot_segs[itr].set_xdata(tofseg)
--> 384     plot_peaks[itr].set_xdata(self.peaks[itr, 0])
    385     plot_peaks[itr].set_ydata(self.peaks[itr, 1])
    387 fig.canvas.draw_idle()

File /asap3/flash/gpfs/pg2/2023/data/11019101/processed/denoising/.cache/pypoetry/virtualenvs/sed-processor-oDlu1RV1-py3.9/lib/python3.9/site-packages/matplotlib/lines.py:1289, in Line2D.set_xdata(self, x)
   1276 """
   1277 Set the data array for x.
   1278 
   (...)
   1286 set_ydata
   1287 """
   1288 if not np.iterable(x):
-> 1289     raise RuntimeError('x must be a sequence')
   1290 self._xorig = copy.copy(x)
   1291 self._invalidx = True

RuntimeError: x must be a sequence
```

Reply:
This is due to a new version of matplotlib, I think. Are you sure you tested with an environment corresponding to the correct lockfile?